core/src/test/java/org/apache/iceberg/actions/TestBinPackStrategy.java

```diff
@@ -138,6 +138,43 @@ public void testGroupingMinInputFilesInvalid() {
         0, Iterables.size(grouped));
   }
 
+  @Test
+  public void testGroupingMinInputFilesAsOne() {
+    RewriteStrategy strategy = defaultBinPack().options(ImmutableMap.of(
+            BinPackStrategy.MIN_INPUT_FILES, Integer.toString(1),
+            BinPackStrategy.MAX_FILE_SIZE_BYTES, Long.toString(3 * MB),
+            RewriteDataFiles.TARGET_FILE_SIZE_BYTES, Long.toString(2 * MB),
+            BinPackStrategy.MIN_FILE_SIZE_BYTES, Long.toString(MB),
+            BinPackStrategy.DELETE_FILE_THRESHOLD, Integer.toString(2)
+    ));
+
+    // single file within a group, with size within threshold and no deletes should be considered as NOOP.
+    Iterable<FileScanTask> testFiles1 = filesOfSize(1);
+    Iterable<List<FileScanTask>> grouped1 = strategy.planFileGroups(testFiles1);
+
+    Assert.assertEquals("Should plan 0 group, as 1 minInputFile with size within threshold and no deletes should be " +
+            "treated as NOOP",
+            0, Iterables.size(grouped1));
+
+    // group with single file with size > targetFileSize should not be planed
+    Iterable<FileScanTask> testFiles2 = filesOfSize(1);
+    Iterable<List<FileScanTask>> grouped2 = strategy.planFileGroups(testFiles2);
+
+    Assert.assertEquals("Should plan 0 group", 0, Iterables.size(grouped2));
+
+    // group with single file with size > maxFileSize should be planed
+    Iterable<FileScanTask> testFiles3 = filesOfSize(4);
+    Iterable<List<FileScanTask>> grouped3 = strategy.planFileGroups(testFiles3);
+
+    Assert.assertEquals("Should plan 1 group", 1, Iterables.size(grouped3));
+
+    // group with single file and with deletes >= deleteThreshold should be planned
+    List<FileScanTask> testFiles4 = Lists.newArrayList();
+    testFiles4.add(MockFileScanTask.mockTaskWithDeletes(MB, 2));
+    Iterable<List<FileScanTask>> grouped4 = strategy.planFileGroups(testFiles4);
+    Assert.assertEquals("Should plan 1 group", 1, Iterables.size(grouped4));
```

Comment:
Should plan 1 group, the data file has delete files and can be re-written without deleted rows.